modules/dkan/dkan_periodic_updates/dkan_periodic_updates.module

```diff
@@ -0,0 +1,459 @@
+<?php
+
+/**
+ * @file
+ * Code for dkan_periodic_updates module.
+ */
+
+/**
+ * Implements hook_permission().
+ */
+function dkan_periodic_updates_permission()
+{
+  return [
+    'administer periodic updates' => [
+      'title' => t('administer periodic updates')
+    ],
+  ];
+}
+
+/**
+ * Implements hook_menu().
+ */
+function dkan_periodic_updates_menu() {
+  $items = [];
+  $items['admin/dkan/periodic-updates'] = [
+    'title' => t('Periodic updates'),
+    'description' => t('Form for uploading periodic updates manifest.'),
+    'page callback' => 'dkan_periodic_updates',
+    'access arguments' => ['administer periodic updates'],
+    'type' => MENU_NORMAL_ITEM,
+  ];
+  $items['admin/dkan/periodic-updates/settings'] = [
+    'title' => t('Settings'),
+    'type' => MENU_DEFAULT_LOCAL_TASK,
+  ];
+  $items['admin/dkan/periodic-updates/status'] = [
+    'title' => t('Status'),
+    'page callback' => 'dkan_periodic_updates_state',
+    'access arguments' => ['administer periodic updates'],
+    'type' => MENU_LOCAL_TASK,
+  ];
+  return $items;
+}
+
+
+/**
+ * Page callback for admin/dkan/periodic-updates.
+ */
+function dkan_periodic_updates() {
+  $output = [];
+  $output['periodic-updates-status'] = drupal_get_form('dkan_periodic_updates_status_form');
+  $output['periodic-updates-manifest'] = drupal_get_form('dkan_periodic_updates_form');
+  return $output;
+}
+
+/**
+ * Form for periodic updates manifest.
+ */
+function dkan_periodic_updates_form($form, &$form_state) {
+  $form['manifest-file'] = [
+    '#type' => 'fieldset',
+    '#title' => 'Manifest for periodic updates',
+  ];
+  $form['manifest-file']['info'] = [
+    '#markup' => '<p>' . t('The manifest file should contain the following column names: resource_id, frequency and file_url. The allowed frequency values are daily, weekly, or monthly. If left blank, the frequency will default to daily.') . '</p>',
+  ];
+  $form['manifest-file']['manifest'] = [
+    '#type' => 'managed_file',
+    '#title' => 'Upload manifest file',
+    '#upload_location' => 'public://manifest/',
+    '#upload_validators' => array(
+      'file_validate_extensions' => array('csv'),
+    ),
+  ];
+  $form['manifest-file']['submit'] = [
+    '#type' => 'submit',
+    '#id' => 'add-manifest-file',
+    '#value' => t('Add file'),
+  ];
+
+  // Current manifest file.
+  $prev_file_name = '- None -';
+  $message = '<p>Current manifest: ' . $prev_file_name . '</p>';
+  $prev_file = variable_get('dkan_periodic_updates_manifest');
+  if ($prev_file) {
+    $file = file_load($prev_file);
+    $prev_file_name = $file->filename;
```

Comment:
Probably not a common occurrence but if the user uploads a manifest file then deletes it via the `admin/content/file` UI then they will get warnings here
```
Notice: Trying to get property of non-object in dkan_periodic_updates_form() (line 87 of /var/www/dkan/modules/dkan/dkan_periodic_updates/dkan_periodic_updates.module).
Notice: Trying to get property of non-object in dkan_periodic_updates_form() (line 88 of /var/www/dkan/modules/dkan/dkan_periodic_updates/dkan_periodic_updates.module).
```
if you wrap lines 87-89 in an `if ($file) {}` statement, we still run into problems because of the **dkan_periodic_updates_manifest** variable, so perhaps we need to check that the variable exists, then confirm the file exists, if not then wipe out the variable.... maybe set the message to 'file was removed' or something...